checks/conciseness/conciseness.py

```diff
@@ -1,47 +1,49 @@
-from aws_lambda_powertools.utilities.typing import LambdaContext
-from aws_lambda_powertools.logging import correlation_paths
-from aws_lambda_powertools import Logger
-from aws_lambda_powertools import Tracer
-from aws_lambda_powertools import Metrics
 from pydantic import ValidationError
 import json
+import openai
 
-from utils.schema import InputSchema, OutputSchema, ErrorSchema
-from utils.prompts import compare_answers_prompt
-from utils.chat_completion import get_chat_completion
-from utils.sample_qa import SAMPLE_QUESTION, SAMPLE_NEW_ANSWER, SAMPLE_OLD_ANSWER
+from utils.check_schema import LLMToolkitStdCheckInputSchema, LLMToolkitStdCheckOutputSchema, ErrorSchema
+from prompt import compare_answers_prompt
 from utils.secret_manager import get_secret
+from utils.logger import get_logger
 
-tracer = Tracer()
-logger = Logger()
-metrics = Metrics(namespace="Powertools")
 
-
-# Enrich logging with contextual information from Lambda
-@logger.inject_lambda_context(correlation_id_path=correlation_paths.API_GATEWAY_REST)
-@tracer.capture_lambda_handler
-def lambda_handler(event: dict, context: LambdaContext) -> dict:
+logger = get_logger(__name__)
+def lambda_handler(event: dict, context: dict) -> dict:
     logger.info(f"Going to run conciseness check on {event=}")
     try:
-        body = InputSchema(**event)
+        body = LLMToolkitStdCheckInputSchema(**event)
         user_prompt, system_prompt = compare_answers_prompt(
             question=body.question,
             old_answer = body.old_answer,
             new_answer=body.new_answer,
             criterion="How concise the answer is"
         )
+
         secrets = get_secret()
```

Comment:
I'm not sure if this is the best way of getting the secrets as I don't think it follows dependency injectable principle as arleif suggested.

You have the SecretsManager class with the Options to pass the the creds and all but while testing this function there is no way to utilise that or to mock that SecretsManager class

Reply:
I will likely shift to Parameter Store https://github.com/trilogy-group/llm-toolkit-templates-evaluation-check/pull/8#discussion_r1302468986
I will do this once I get the basic functionality working